src/core/analysis/mcmc/Mcmcmc.cpp

```diff
@@ -1136,14 +1136,14 @@ void Mcmcmc::setHeatsInitial(const std::vector<double>& ht)
         throw RbException()<<"Heat vector contains "<<ht.size()<<" heats, but there are "<<num_chains<<" chains.";
 
     // 2. Check that the heats are non-zero
-    for(int i=0; i<int(ht.size()); i++)
+    for(size_t i=0; i<ht.size(); i++)
     {
         if (not (ht[i] > 0.0))
             throw RbException()<<"Heat "<<i+1<<" is "<<ht[i]<<", but should be > 0";
     }
 
     // 3. Check that the heats are sorted with the largest heat first (== smallest temperature first).
-    for(int i=0;i<int(ht.size())-1;i++)
+    for(size_t i=0;i<ht.size()-1;i++)
```

Comment:
Also here.

Reply:
Here you can do something like `std::int64_t(ht.size())`, and that avoids the issue with `-1` wrapping around and becoming a large, positive number.

Reply:
So here I think we can avoid casting to a signed type by just avoiding subtraction. We can check `i+1 <ht.size()`, which might make more sense anyway if we are trying to check that `ht[i+1]` is valid.